### PR TITLE
SQL part 1 exercise

### PR DESCRIPTION
**Pull request by gilRoz23**

### CODE REVIEW

File: src/main/resources/db/migration/V202408091500__create_owners_table.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+create table owners
+(
+    id serial primary key,
+    name text,
+    company_id int,
+    employee_id int
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091500__create_owners_table.sql:6]: employee_id in bob is type string

File: src/main/resources/db/migration/V202408091500__create_owners_table.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+)
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091500__create_owners_table.sql:7]: missing index

File: src/main/resources/db/migration/V202408091400__create_pets_table.sql [hunk 3]
```diff
@@ -0,0 +1,18 @@
+create table pets
+(
+    id SERIAL primary key ,
+    name varchar(255),
+    type varchar(255),
+    company_id integer,
+    date_of_arrival date
+);
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091400__create_pets_table.sql:8]: missing index

File: src/main/resources/db/migration/V202408091400__create_pets_table.sql [hunk 4]
```diff
@@ -0,0 +1,18 @@
+create table pets
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091400__create_pets_table.sql:1]: the name of the file should be the date it was created

File: src/main/resources/db/migration/V202408091500__create_owners_table.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+create table owners
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091500__create_owners_table.sql:1]: same here

File: src/main/resources/db/migration/V202408091400__create_pets_table.sql [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    date_of_arrival date
+);
+
+INSERT INTO pets VALUES (1,"Murphy","dog",1,"2024-09-08");
+INSERT INTO pets VALUES(2,"Pepe","dog",1,"2024-09-09");
+INSERT INTO pets VALUES(1,"Garfield","cat",2,"2024-09-08");
+
+-- SELECT * from pets where type = 'dog';
+
+delete from pets where name='Garfield'
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091400__create_pets_table.sql:16]: Delete pet by id

File: src/main/resources/db/migration/V202408091400__create_pets_table.sql [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+-- SELECT * from pets where date_of_arrival<'2023-09-08';
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091400__create_pets_table.sql:18]: this is not the query - Get the pets that arrived more than a year ago
this query brings all pets that arrived before '09-08-2023'

File: src/main/resources/db/migration/V202408091400__create_pets_table.sql [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    id SERIAL primary key ,
+    name varchar(255),
+    type varchar(255),
+    company_id integer,
+    date_of_arrival date
+);
+
+INSERT INTO pets VALUES (1,"Murphy","dog",1,"2024-09-08");
+INSERT INTO pets VALUES(2,"Pepe","dog",1,"2024-09-09");
+INSERT INTO pets VALUES(1,"Garfield","cat",2,"2024-09-08");
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091400__create_pets_table.sql:12]: it wont work the same id for Murphy and Garfield

File: src/main/resources/db/migration/V202408091400__create_pets_table.sql [hunk 9]
```diff
@@ -0,0 +1,18 @@
+create table pets
+(
+    id SERIAL primary key ,
+    name varchar(255),
+    type varchar(255),
+    company_id integer,
+    date_of_arrival date
+);
+
+INSERT INTO pets VALUES (1,"Murphy","dog",1,"2024-09-08");
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091400__create_pets_table.sql:10]: put these queries in different file

**Review by mayhuberman — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/V202408091400__create_pets_table.sql [hunk 10]
```diff
@@ -0,0 +1,20 @@
+create table pets
+(
+    id SERIAL primary key ,
+    name varchar(255),
+    type varchar(255),
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091400__create_pets_table.sql:5]: should be not null
think what fields should be not null

File: src/main/resources/db/migration/V202408091400__create_pets_table.sql [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    company_id integer,
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091400__create_pets_table.sql:6]: company_id in bob is bigint

File: src/main/resources/db/migration/V202408091400__create_pets_table.sql [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+    date_of_arrival date default current_date
+);
+
+CREATE INDEX idx_company_id ON pets (company_id);
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091400__create_pets_table.sql:10]: add the name of the table to the index name

File: src/main/resources/db/migration/V202408091500__create_owners_table.sql [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+(
+    id serial primary key,
+    name varchar,
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091500__create_owners_table.sql:4]: same here think what fields should be not null

File: src/main/kotlin/com/hibob/AcademyApplication.kt [hunk 14]
```diff
@@ -4,7 +4,6 @@ import org.springframework.boot.autoconfigure.SpringBootApplication
 import org.springframework.boot.autoconfigure.flyway.FlywayAutoConfiguration
 import org.springframework.boot.autoconfigure.security.servlet.SecurityAutoConfiguration
 import org.springframework.boot.autoconfigure.security.servlet.UserDetailsServiceAutoConfiguration
-import org.springframework.boot.autoconfigure.thymeleaf.ThymeleafAutoConfiguration
```
[the remark this answers is not in the document]
Comment (mayhuberman) [reply src/main/kotlin/com/hibob/AcademyApplication.kt:7]: no need for it on this pr

File: src/main/resources/db/migration/V202408091500__create_owners_table.sql [hunk 15]
```diff
@@ -0,0 +1,10 @@
...
+    company_id int,
+    employee_id varchar
+)
+
+-- CREATE INDEX idx_company_id ON owners (company_id);
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091500__create_owners_table.sql:9]: missing index

**Review by mayhuberman — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/V202408091500__create_owners_table.sql [hunk 16]
```diff
@@ -0,0 +1,8 @@
+create table owners
+(
+    id serial primary key NOT NULL,
+    name varchar NOT NULL,
+    company_id int NOT NULL,
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202408091500__create_owners_table.sql:5]: bigint

**Review by rakreshet — APPROVED** [review]

**Pull request merged**